modules/ROOT/pages/syntax-quick-reference.adoc

```diff
@@ -0,0 +1,1290 @@
+====
+
+.Inline image macro with positioning role
+[#ex-image-role]
+----
+include::macros:example$image.adoc[tag=in-role]
+----
+
+.View result of <<ex-image-role>>
+[%collapsible.result]
+====
+include::macros:example$image.adoc[tag=qr-role]
+====
+
+.Embedded
+----
+include::macros:example$image.adoc[tag=data]
+----
+
+When the `data-uri` attribute is set, all images in the document--including admonition icons--are embedded into the document as {url-data-uri}[data URIs].
+You can also pass it as a command line argument using `-a data-uri`.
+
+== Videos
+
+.Block video macro
+----
+include::macros:example$video.adoc[tag=base]
+
+include::macros:example$video.adoc[tag=attr]
+----
+
+.Embedded Youtube video
+----
+include::macros:example$video.adoc[tag=you]
+----
+
+.Embedded Vimeo video
+----
+include::macros:example$video.adoc[tag=vim]
+----
+
+You can control the video settings using xref:macros:audio-and-video.adoc[additional attributes and options] on the macro.
+
+== Source Code
+
+.Inline as monospace only
+[#ex-inline-code]
+----
+include::text:example$text.adoc[tag=b-mono-code]
+----
+
+.View result of <<ex-inline-code>>
+[%collapsible.result]
+====
+include::text:example$text.adoc[tag=b-mono-code]
+====
+
+.Inline as monospace and literal (i.e., literal monospace)
+[#ex-inline-literal]
+----
+include::pass:example$pass.adoc[tag=backtick-plus]
+----
+
+.View result of <<ex-inline-literal>>
+[%collapsible.result]
+====
+include::pass:example$pass.adoc[tag=backtick-plus]
+====
+
+.Literal and monospace line
+[#ex-literal-line]
+----
+include::verbatim:example$literal.adoc[tag=b-imp-code]
+----
+
+.View result of <<ex-literal-line>>
+[%collapsible.result]
+====
+include::verbatim:example$literal.adoc[tag=b-imp-code]
+====
+
+.Literal block
+[#ex-literal-block]
+----
+include::verbatim:example$literal.adoc[tag=b-block]
+----
+
+.View result of <<ex-literal-block>>
+[%collapsible.result]
+====
+include::verbatim:example$literal.adoc[tag=b-block]
+====
+
+.Listing block with title
+[#ex-listing]
+------
+include::verbatim:example$listing.adoc[tag=qr-listing]
+------
+
+.View result of <<ex-listing>>
+[%collapsible.result]
+====
+[caption="Listing 1. "]
+include::verbatim:example$listing.adoc[tag=qr-listing]
+====
+
+.Source block with title and syntax highlighting
+[#ex-highlight]
+------
+.Some Ruby code
+include::verbatim:example$source.adoc[tag=src-base]
+------
+
+.View result of <<ex-highlight>>
+[%collapsible.result]
+====
+[caption="Listing 1. "]
+.Some Ruby code
+include::verbatim:example$source.adoc[tag=src-base]
+====
+
+.Enable source highlighting
+[IMPORTANT]
+====
+xref:verbatim:source-highlighter.adoc[Source highlighting] is enabled by setting the `source-highlighter` attribute in the document header or passed as an argument.
+
+[source]
+----
+:source-highlighter: rouge
+----
+
+The built-in values are `coderay`, `highlightjs`, `rouge`, and `pygments`.
+====
+
+.Source block with callouts
```

Comment:
question: Should we mention the `<.>` syntax, maybe just a quick note?

Reply:
It should probably be added at some point, though it isn't documented in the main docs yet either. It depends on whether we're adding new syntax to the quick ref while we're migrating it or only correcting existing examples that are wrong/outdated/boring-bad?

Reply:
I agree, that needs to be a follow-up. It's a newer feature and needs to first be documented.

Reply:
Sounds good :+1: